main.cc

```diff
@@ -1257,11 +1257,11 @@ int main(int ac, char** av) {
                 }
                 bool alternator_enforce_authorization = cfg->alternator_enforce_authorization();
                 with_scheduling_group(dbcfg.statement_scheduling_group,
-                        [addr, alternator_port, alternator_https_port, creds = std::move(creds), alternator_enforce_authorization] () mutable {
+                        [addr, alternator_port, alternator_https_port, creds = std::move(creds), alternator_enforce_authorization, cfg] () mutable {
                     return alternator_server.invoke_on_all(
-                            [addr, alternator_port, alternator_https_port, creds = std::move(creds), alternator_enforce_authorization] (alternator::server& server) mutable {
+                            [addr, alternator_port, alternator_https_port, creds = std::move(creds), alternator_enforce_authorization, cfg] (alternator::server& server) mutable {
                         auto& ss = service::get_local_storage_service();
-                        return server.init(addr, alternator_port, alternator_https_port, creds, alternator_enforce_authorization, &ss.service_memory_limiter());
+                        return server.init(addr, alternator_port, alternator_https_port, creds, alternator_enforce_authorization, &ss.service_memory_limiter(), cfg->max_concurrent_requests_per_shard);
```

Comment:
A question - is "max_concurrent_requests_per_shard" the same number for CQL and Alternator? Maybe this is good enough for now as an emergency break for support people. But in general there is no reason to assume that it needs to be the same number. Moreover, there is no reason to assume that the limit on reads and writes should be the same, or non-lwt vs lwt writes, and so on :-(

Reply:
It was discussed in https://github.com/scylladb/scylla/issues/7072, https://github.com/scylladb/scylla/issues/7279 and probably a couple of other places as well -- it's just a starter to be used for emergency situations. Eventually, it would be ideal for this parameter to be automatically adjusted, and it should not be just a single value, but rather split into multiple areas - reads, writes, lwts, etc.

As for CQL vs alternator (vs redis), I also mentioned here https://github.com/scylladb/scylla/pull/7279#issuecomment-700529918 (we **do** have to many threads for this discussion...), that for now I assume that people don't really use CQL and alternator on the same cluster for production, and if they do, the limit is good enough for emergencies. Later, it would of course make sense to split the parameter per-client, if the need arises.